src/test_workflow/benchmark_test/benchmark_test_cluster.py

```diff
@@ -0,0 +1,158 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+
+import json
+import logging
+import os
+import subprocess
+from contextlib import contextmanager
+from typing import Any, Generator
+
+import requests
+from requests.auth import HTTPBasicAuth
+from retry.api import retry_call  # type: ignore
+
+from manifests.bundle_manifest import BundleManifest
+from test_workflow.benchmark_test.benchmark_args import BenchmarkArgs
+
+
+class BenchmarkTestCluster:
+    manifest: BundleManifest
+    work_dir: str
+    current_workspace: str
+    args: BenchmarkArgs
+    output_file: str
+    params: str
+    is_endpoint_public: bool
+    cluster_endpoint: str
+    cluster_endpoint_with_port: str
+
+    """
+    Represents a performance test cluster. This class deploys the opensearch bundle with CDK. Supports both single
+    and multi-node clusters
+    """
+
+    def __init__(
+            self,
+            bundle_manifest: BundleManifest,
+            config: dict,
+            args: BenchmarkArgs,
+            current_workspace: str
+    ) -> None:
+        self.manifest = bundle_manifest
+        self.current_workspace = current_workspace
+        self.args = args
+        self.output_file = "output.json"
+        role = config["Constants"]["Role"]
+        params_dict = self.setup_cdk_params(config)
+        params_list = []
+        for key, value in params_dict.items():
+            print(key, value)
+            if value:
+                if key == 'additionalConfig':
+                    params_list.append(f" -c {key}=\'{value}\'")
+                else:
+                    params_list.append(f" -c {key}={value}")
+        role_params = (
+            f" --require-approval=never --plugin cdk-assume-role-credential-plugin"
+            f" -c assume-role-credentials:writeIamRoleName={role} -c assume-role-credentials:readIamRoleName={role} "
+        )
+        self.params = "".join(params_list) + role_params
+        self.is_endpoint_public = False
+        self.cluster_endpoint = None
+        self.cluster_endpoint_with_port = None
+        self.stack_name = 'opensearch-infra-stack-' + self.args.stack_suffix + '-' + self.manifest.build.id + '-' + self.manifest.build.architecture
+
+    def start(self) -> None:
+        # os.chdir(self.work_dir)
+        command = f"npm install && cdk deploy \"*\" {self.params} --outputs-file {self.output_file}"
+
+        logging.info(f'Executing "{command}" in {os.getcwd()}')
+        subprocess.check_call(command, cwd=os.getcwd(), shell=True)
+        with open(self.output_file, "r") as read_file:
+            load_output = json.load(read_file)
+            print(load_output[self.stack_name])
```

Comment:
Use `logging` if you need to print something.

Reply:
Not required, removed.